csharp/src/Ice/ConnectionFactory.cs

```diff
@@ -199,32 +192,11 @@ public void SetRouterInfo(RouterInfo routerInfo)
                 //
                 for (int i = 0; i < endpoints.Count; ++i)
                 {
-                    Endpoint endpoint = endpoints[i];
-
-                    //
-                    // Modify endpoints with overrides.
-                    //
-                    if (_communicator.OverrideTimeout != null)
-                    {
-                        endpoint = endpoint.NewTimeout(_communicator.OverrideTimeout.Value);
-                    }
-
-                    //
-                    // The ConnectionI object does not take the compression flag of
-                    // endpoints into account, but instead gets the information
-                    // about whether messages should be compressed or not from
-                    // other sources. In order to allow connection sharing for
-                    // endpoints that differ in the value of the compression flag
-                    // only, we always set the compression flag to false here in
-                    // this connection factory.
-                    //
-                    endpoint = endpoint.NewCompressionFlag(false);
-
                     foreach (ICollection<Connection> connections in _connectionsByConnector.Values)
                     {
                         foreach (Connection connection in connections)
                         {
-                            if (connection.Endpoint == endpoint)
+                            if (connection.Endpoint == endpoints[i])
```

Comment:
can we use a foreach for the most outer loop?

Reply:
Yes, I'll fix